Enabling guest agent for new VM crashes plugin

When creating a new VM from an unprovisioned ISO (from scratch) using the argument `agent = 1 ` in the `proxmox_vm_qemu` resource, the plugin breaks. Even though the VM is created, it becomes independent of Terraform and `terraform destroy` will say that there is nothing to destroy.

```
│ Error: Request cancelled
│ 
│   with proxmox_vm_qemu.kube["k3s-master-1"],
│   on instances.tf line 1, in resource "proxmox_vm_qemu" "kube":
│    1: resource "proxmox_vm_qemu" "kube" {
│ 
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-proxmox_v3.0.1-rc3 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 55 [running]:
github.com/Telmate/terraform-provider-proxmox/v2/proxmox.getPrimaryIP(0xc000278780, 0xc000bf2048, 0xc000bf1950?, 0x0?, {0xc000ed5c00?, 0x1c?, 0x13a5340?}, 0x5, 0x0, 0x1, ...)
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/resource_vm_qemu.go:1933 +0xf26
github.com/Telmate/terraform-provider-proxmox/v2/proxmox.initConnInfo({0xc0000aa2a0?, 0x0?}, 0xc0002f4080, 0xc000bf1ee8?, 0xc000edaf10?, 0xc000bf2048, 0xc000278780, 0x1f4?)
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/resource_vm_qemu.go:1869 +0x825
github.com/Telmate/terraform-provider-proxmox/v2/proxmox.resourceVmQemuRead({0xe47340, 0xc00069dc70}, 0xc0002f4080, {0xbb8f80?, 0xc000263ae0})
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/resource_vm_qemu.go:1440 +0x92e
github.com/Telmate/terraform-provider-proxmox/v2/proxmox.resourceVmQemuCreate({0xe47340, 0xc00069dc70}, 0xc0002f4080, {0xbb8f80?, 0xc000263ae0?})
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/resource_vm_qemu.go:1094 +0x2ee5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0005ccc40, {0xe47298, 0xc000ae6d50}, 0xd?, {0xbb8f80, 0xc000263ae0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:806 +0x11b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0005ccc40, {0xe47298, 0xc000ae6d50}, 0xc000162c30, 0xc000135280, {0xbb8f80, 0xc000263ae0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:937 +0xa89
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000390000, {0xe47298?, 0xc000ae6c60?}, 0xc000a86370)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:1153 +0xdbc
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0002500a0, {0xe47298?, 0xc000ae6240?}, 0xc00069c1c0)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:865 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xce8520?, 0xc0002500a0}, {0xe47298, 0xc000ae6240}, 0xc000134300, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:518 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001a5000, {0xe47298, 0xc000ae61b0}, {0xe4b2f8, 0xc000248780}, 0xc00041a120, 0xc00059ccf0, 0x1397d78, 0x0)
        google.golang.org/grpc@v1.63.2/server.go:1369 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc0001a5000, {0xe4b2f8, 0xc000248780}, 0xc00041a120)
        google.golang.org/grpc@v1.63.2/server.go:1780 +0x1016
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 36
        google.golang.org/grpc@v1.63.2/server.go:1030 +0x135

Error: The terraform-provider-proxmox_v3.0.1-rc3 plugin crashed!
```

I understand that enabling the guest agent won't work in a VM where `qemu_guest_agent` isn't installed, but I don't think this is expected behaviour either? 

Thank you.

--- a/proxmox/resource_vm_qemu.go
+++ b/proxmox/resource_vm_qemu.go
@@ -1400,6 +1400,7 @@ func resourceVmQemuRead(ctx context.Context, d *schema.ResourceData, meta interf
 	}
 	if err == nil && vmState["status"] == "running" {
 		log.Printf("[DEBUG] VM is running, checking the IP")
+		// TODO when network interfaces are reimplemented check if we have an interface before getting the connection info
 		diags = append(diags, initConnInfo(d, client, vmr, config)...)
 	} else {
 		// Optional convenience attributes for provisioners
@@ -1870,11 +1871,18 @@ func getPrimaryIP(config *pxapi.ConfigQemu, vmr *pxapi.VmRef, client *pxapi.Clie
 	// get all information we can from qemu agent until the timer runs out
 	if ciAgentEnabled {
 		var waitedTime int
+		// TODO rework this logic when network interfaces are properly handled in the SDK
 		vmConfig, err := client.GetVmConfig(vmr)
 		if err != nil {
 			return primaryIPs{}, diag.FromErr(err)
 		}
-		net0MacAddress := macAddressRegex.FindString(vmConfig["net0"].(string))
+		var primaryMacAddress string
+		for i := 0; i < 16; i++ {
+			if _, ok := vmConfig["net"+strconv.Itoa(i)]; ok {
+				primaryMacAddress = macAddressRegex.FindString(vmConfig["net"+strconv.Itoa(i)].(string))
+				break
+			}
+		}
 		for time.Now().Before(endTime) {
 			var interfaces []pxapi.AgentNetworkInterface
 			interfaces, err = vmr.GetAgentInformation(client, false)
@@ -1888,7 +1896,7 @@ func getPrimaryIP(config *pxapi.ConfigQemu, vmr *pxapi.VmRef, client *pxapi.Clie
 				if len(interfaces) > 0 { // agent returned some information
 					log.Printf("[INFO][getPrimaryIP] QEMU Agent interfaces found: %v", interfaces)
 					logger.Debug().Int("vmid", vmr.VmId()).Msgf("QEMU Agent interfaces found: %v", interfaces)
-					conn = conn.parsePrimaryIPs(interfaces, net0MacAddress)
+					conn = conn.parsePrimaryIPs(interfaces, primaryMacAddress)
 					if conn.hasRequiredIP() {
 						return conn.IPs, diag.Diagnostics{}
 					}

--- a/proxmox/resource_vm_qemu.go
+++ b/proxmox/resource_vm_qemu.go
@@ -1394,11 +1394,12 @@ func resourceVmQemuRead(ctx context.Context, d *schema.ResourceData, meta interf
 	}
 
 	vmState, err := client.GetVmState(vmr)
-	log.Printf("[DEBUG] VM status: %s", vmState["status"])
-	if err == nil {
-		d.Set("vm_state", vmState["status"])
+	if err != nil {
+		return diag.FromErr(err)
 	}
-	if err == nil && vmState["status"] == "running" {
+	log.Printf("[DEBUG] VM status: %s", vmState["status"])
+	d.Set("vm_state", vmState["status"])
+	if vmState["status"] == "running" {
 		log.Printf("[DEBUG] VM is running, checking the IP")
 		// TODO when network interfaces are reimplemented check if we have an interface before getting the connection info
 		diags = append(diags, initConnInfo(d, client, vmr, config)...)
@@ -1410,9 +1411,6 @@ func resourceVmQemuRead(ctx context.Context, d *schema.ResourceData, meta interf
 		diags = append(diags, diag.FromErr(err)...)
 		err = d.Set("ssh_port", nil)
 		diags = append(diags, diag.FromErr(err)...)
-	}
-	if err != nil {
-		return diag.FromErr(err)
 	}
 
 	logger.Debug().Int("vmid", vmID).Msgf("[READ] Received Config from Proxmox API: %+v", config)

